### PR TITLE
Celotool --helmdryrun cmd line option

### LINKED ISSUE #1331: Add dry-run functionality to celotool's deploy upgrade commands
### Expected Behavior

When preparing to conduct an upgrade, you should be able to see the changes that the upgrade would do before committing to do it.  Specifically, the changes in the values (using helm diff) and the commands that it would run.

### Current Behavior

Currently it shows the results of running the upgrade and the command that it runs, but does not use diff to show it in a user-friendly way, and there is no option to do dry run without doing the upgrade

### System information

Run `geth version`

### PR DESCRIPTION
**Pull request by jcortejoso**
### Description

Added the cmd option `--helmdryrun` for `celotool deploy initial` and `celotool deploy upgrade`. Adding this option simulates the execution of the helm upgrade, or shows the helm parameters that would be used in the initial deployment. This option does not include the simulation of other operations outside helm like creation of sql instances or static ips, so cannot be used as an stealth command for any case.

### Other changes

- Removed two deprecated env variables related with geth exporter (not longer used).
- Increased the minimum Helm version required to `3.4` due to some limitations of the helm command used for showing the upgrade difference.

### Tested

Tested in testnets.

### Related issues

- Fixes celo-org/celo-blockchain#1331

### Backwards compatibility

No problem detected.

### CODE REVIEW

**Comment by oneeman** [comment]
Is this PR being superseded by #6831 or is the idea to first merge this one and then later merge that one?

**Comment by jcortejoso** [comment]
@oneeman I'd merge this one first and then  #6831 so we can have two different (and smaller) commits, as this one includes more code changes that are not needed for the other PR.

File: packages/celotool/src/cmds/deploy/initial/testnet.ts [hunk 1]
```diff
@@ -41,8 +41,9 @@ export const handler = async (argv: TestnetInitialArgv) => {
   await createStaticIPs(argv.celoEnv)
```
Comment (trianglesphere) [inline packages/celotool/src/cmds/deploy/initial/testnet.ts:41]: Should this only be done if it is not a dry run?

**Review by trianglesphere — COMMENTED** [review]
I just have a couple questions about the scope of which commands will have the `--helmdryrun` option.
We have a lot of deploys that are through helm charts (which automatically get this command line option with full effect). The deploys I'd like to have listed are the ones with other configuration. We've done this for `testnet`, but there are others with configuration beyond a helm chart.
`deploy initial`
- voting bot
- tracer too
- load test
- fullnodes
- blockscout
- blockchain api

`deploy upgrade`
- blockscout
- fullnodes
- load test

Note, this is not related, but in `deploy destroy celostats` with `createClusterIfNotExists` which is probably not what we want to do.

We also do not apply `--helmdryrun` to `deploy destroy`, but that feels fine to me.
I don't think we need to enable `--helmdryrun` for each command I listed, but I do think we should at least note that `helmdryrun` is not quite the same as a `dryrun` for certain deploys.

File: packages/celotool/src/cmds/deploy/initial/testnet.ts (hunk 1 above)
Comment (jcortejoso) [reply packages/celotool/src/cmds/deploy/initial/testnet.ts:41]: The IPs are required values for the helm chart, so we'd need to pass them (or alternatively, submit some fake address if they do not exist).

**Comment by jcortejoso** [comment]
@trianglesphere the `--helmdryrun` should be available to any other command that uses the functions `installGenericHelmChart` and `upgradeGenericHelmChart`. A quick look to the code seems that all the packages are using this functions with the exception of `eksportisto`, `tracer-tool`, and `transaction-metrics-exporter`. I am not sure if these packages have any peculiarity so they cannot use the generic helm functions, but I would suggest creating another issue to check if is possible to use the same methods for those packages, and if not possible then we can add the `dry-run` functionality to those packages particularly.

I prefered to name the option `--helmdryrun` to make it more explicit that the dry-run affects directly to helm but maybe other changes done by the celotool command can be executed anyway (like for example blockscout, it creates the database and some policies outside the helm package that are still run when running this option).
If we want to extend the `--helmdryrun` to a `--dry-run` option we should check case by case an check if:
- There is a dependency with the helm values and the steps outside helm (typically it is).
- Refactor the package to avoid doing things outside helm, or refactor so we take into account the dry-run option.

I'd consider this is out of scope this issue and we can open a successive PRs in relation with the changes to extend this functionality.

**Review by trianglesphere — APPROVED** [review]
Ok, I've made a ticket to track my concerns here: https://github.com/celo-org/celo-labs/issues/789
The main thing I'm worried about is using `--helmdryrun` on a deploy that has some custom logic and some helm chart logic, but `--helmdryrun` does make it clear that it's only a dry run for helm and that there's a bit of caveat emptor.

**Pull request merged**